src/garage/tf/samplers/ray_batched_sampler.py

```diff
@@ -0,0 +1,71 @@
+"""Ray Sampler, for tensorflow algorithms.
+
+Currently the same as garage.samplers.RaySampler but includes
+support for Tensorflow sessions
+"""
+import ray
+import tensorflow as tf
+
+from garage.sampler import RaySampler, SamplerWorker
+
+
+class RaySamplerTF(RaySampler):
+    """Ray Sampler, for tensorflow algorithms.
+
+    Currently the same as garage.samplers.RaySampler
+
+    Args:
+        - Same as garage.samplers.RaySampler
+    """
+
+    def __init__(self,
+                 algo,
+                 env,
+                 seed,
+                 should_render=False,
+                 num_processors=None):
+        super().__init__(
+            algo,
+            env,
+            seed,
+            should_render=False,
+            num_processors=None,
+            sampler_worker_cls=SamplerWorkerTF)
+
+    def shutdown(self):
+        """Shuts down the worker."""
+        temp = []
+        for worker in self._all_workers.values():
+            temp.append(worker.shutdown.remote())
+        ray.get(temp)
+        ray.shutdown()
+
+
+class SamplerWorkerTF(SamplerWorker):
+    """Sampler Worker for tensorflow on policy algorithms.
+
+    - Same as garage.samplers.SamplerWorker, except it
+    initializes a tensorflow session, because each worker
+    is in a separate process.
+    """
+
+    def __init__(self,
+                 worker_id,
+                 env,
+                 agent,
+                 seed,
+                 max_path_length,
+                 should_render=False):
+        self.sess = tf.Session()
+        self.sess.__enter__()
```

Comment:
Is this necessary?

Reply:
Yeah I have to create a new session for all of the new workers, since they sit on separate processes

Reply:
Oh I am just wondering if we have to call `__enter__` here.

Reply:
Well we have to enter the session at some point in order to sample from the policy. We can do this with with statements that open and close the policy for us, or since each of the sampler workers won’t have any other sessions running on them, we can enter the session upon initialization, and exit when we’re shutting down the workers.